regular_expressions_intermediate/regular_expressions_intermediate.md

```diff
@@ -0,0 +1,540 @@
+| Speed 2   | Cruise Control Teenage Mutant Ninja Turtles II | 
+| The Secret Of The Ooze Tomb Raider | The Cradle Of Life |
+
+It seems like the regular expression parser isn't stopping at the end of a string and restarting at the beginning of a string.  You assist your collague by adding syntax to their regular expression such that the pattern must start at the beginning of a string and end at the end of a string.  This is what you write:  
+
+`^([A-Za-z0-9\s]+):([A-Za-z0-9\s]+)$`
+
+When you [try this in Regex 101](https://regex101.com/r/Ld3HWY/1) you get exactly what was intended:
+
+![Using a start of line anchor, capturing groups, and end of line anchor, we correctly segment the movie titles into their main and subtitles.](media/capturing_with_anchors.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width:600px;" -->
+
+<div class = "options">
+<b style="color: rgb(var(--color-highlight));">Another option</b><br>
+
+There are other ways to solve your colleague's problem.  You may have realized that the `\s` that allowed for spaces in titles and subtitles was too broad -- `\s` allows for spaces, tabs, carriage returns, and new lines.  You could also have rewritten their regex by changing the `\s` to just be a literal space, like this: `([A-Za-z0-9 ]+):([A-Za-z0-9 ]+)`
+
+In regular expressions, there are often many ways to accomplish the same goal!
+
+</div>
+
+
+## Boundaries 
+
+Boundaries are similar to anchors in that they allow you to think in advanced ways about where your pattern is positioned.  For example, let's consider **words** and **word boundary**.  
+
+Word characters, expressed in regex as `\w`, are (generally, and this can depend a bit by what flavor of regular expression engine you're using) letters, numbers, and underscores (`_`).  Where word characters meet non-word characters, there's a word boundary, indicated in regex as `\b`.
+
+In the string `subject AB_123 has 10 lesions.`, there are five word boundaries. Between the words and after the last word are non-word characters (four spaces and one period):
+
+*  The point between the "t" in "subject" and the space following it is a word boundary and can be expressed as `\b`.  
+* The same is true for the boundary between "3" and the space following it.
+* Also there's a word boundary between the "s" in has and the space following it.
+* The point between "0" and the space following it, is also a word boundary.
+* Finally, between the "s" in "lesions" and the period following it is our final word boundary.
+
+Word boundaries are useful in case you're looking for patterns that match the start or end of a word.  For example, what if you're looking in a text for "ome" words -- mentions of "microbiome", "genome", "transcriptome", "exome", "proteome", "trisome", "microsome" and similar words?  [If you look just for the pattern `ome`](https://regex101.com/r/gk7BYK/1), you will also match "women", "foment", "moment", "omega" and so on.  
```

Comment:
So many commas!!!
```suggestion
Word boundaries are useful in case you're looking for patterns that match the start or end of a word.  For example, what if you're looking in a text for "ome" words -- mentions of "microbiome," "genome," "transcriptome," "exome," "proteome," "trisome," "microsome," and similar words?  [If you look just for the pattern `ome`](https://regex101.com/r/gk7BYK/1), you will also match "women," "foment," "moment," "omega," and so on.  
```